src/main/java/org/fundacionjala/coding/marines/katas/EANValidator.java

```diff
@@ -0,0 +1,47 @@
+package org.fundacionjala.coding.marines.katas;
+
+/** Kata https://www.codewars.com/kata/55563df50dda59adf900004d . **/
+public final class EANValidator {
+
+    /** Private default constructor. **/
+    private EANValidator() {
+    }
+
+    /**
+     * This method get the EAN checksum.
+     * @param eanCode ean code of 12 chars.
+     * @return checksum
+     */
+    public static String getChecksum(final String eanCode) {
+        final int ten = 10;
+        final int three = 3;
+        boolean flipflop = true;
+        int checksum = 0;
+        for (final char ch : eanCode.toCharArray()) {
+            checksum += Integer.valueOf(String.valueOf(ch)) * (flipflop ? 1
+                    : three);
+            flipflop = !flipflop;
+        }
+        checksum = ten - (checksum % ten);
+        return String.valueOf(checksum == ten ? 0 : checksum);
+    }
+
+    /**
+     * This method get the ean code of 13 chars.
+     * @param eanCode12 ean code of 12 chars (without checksum)
+     * @return ean code of 13 chars.
+     */
+    public static String getEanCode(final String eanCode12) {
+        return eanCode12.concat(getChecksum(eanCode12));
+    }
+
+    /**
+     * This method validate an ean code of 13 chars.
+     * @param eanCode ean code of 13 chars.
+     * @return boolean tru if it's valid and false if not.
+     */
+    public static boolean validate(final String eanCode) {
```

Comment:
This may be refactored like this.
```java
    private static final int TEN = 10;
    private static final int THREE = 3;

    public static boolean validate(final String eanCode) {
        boolean flipflop = true;
        int sum = 0;
        char[] eanCode12Chars = eanCode.substring(0, 12).toCharArray();
        for (final char ch : eanCode12Chars) {
            sum += Character.getNumericValue(ch) * (flipflop ? 1 : THREE);
            flipflop = !flipflop;
        }
        int checksum = sum % TEN == 0 ? 0 : TEN - (sum % TEN) ;
        return checksum == Integer.parseInt(eanCode.substring(12));
    }
```